src/flow.c

```diff
@@ -1131,7 +1131,7 @@ uint8_t FlowGetDisruptionFlags(const Flow *f, uint8_t flags)
     TcpSession *ssn = f->protoctx;
     TcpStream *stream = flags & STREAM_TOSERVER ? &ssn->client : &ssn->server;
 
-    if (stream->flags & STREAMTCP_STREAM_FLAG_DEPTH_REACHED) {
+    if (stream->flags & (STREAMTCP_STREAM_FLAG_DEPTH_REACHED | STREAMTCP_FLAG_APP_LAYER_DISABLED)) {
```

Comment:
Q: what happens if UDP parser gets into an error state?

Reply:
Glad you ask.
I started discussing it in redmine cf https://redmine.openinfosecfoundation.org/issues/4318#note-4

Let's continue here :
I did not find any such thing as an error state for UDP (which seems relevant to the nature of UDP and TCP)

Furthermore, if I have this scenario over UDP :
- first request ok, creating first transaction
- second request, invalid returning a parser error
- first response, completing the first transaction

We see that having an error on the second request should not make us clean the first request... right ?

Reply:
I'm not sure. While I understand the out of order nature, don't many protocols have an "ID" even for UDP to match a request against the correct response even if they arrived weirdly? From what I understand that should be considered a complete tx, I know that all protocols do not follow this.. Do I make any sense to you?

Reply:
> I'm not sure. While I understand the out of order nature, don't many protocols have an "ID" even for UDP to match a request against the correct response even if they arrived weirdly? 

Yes, even so...

> From what I understand that should be considered a complete tx, I know that all protocols do not follow this.. Do I make any sense to you?

Yes, even so :
I do not think it makes sens to have such a thing an error state for protocols over UDP.
Or maybe, just dismiss the app-layer protocol detected if we only get errors out of  `AppLayerHandleUdp` ?

Reply:
@jasonish do you have an opinion on this ? (error state for protocols over UDP)